Assets/Scripts/Models/World.cs

```diff
@@ -1063,24 +1064,42 @@ void ReadXml_Characters(XmlReader reader)
             {
                 int x = int.Parse(reader.GetAttribute("X"));
                 int y = int.Parse(reader.GetAttribute("Y"));
+                Character character;
                 if(reader.GetAttribute("r") != null)
                 {
                     float r = float.Parse(reader.GetAttribute("r"));
                     float b = float.Parse(reader.GetAttribute("b"));;
                     float g = float.Parse(reader.GetAttribute("g"));;
                     Color color = new Color(r, g, b, 1.0f);
-                    Character c = CreateCharacter(tiles[x, y], color);
-                    c.name = reader.GetAttribute("name");
-                    c.ReadXml(reader);
+                    character = CreateCharacter(tiles[x, y], color);
+                    character.name = reader.GetAttribute("name");
```

Comment:
You could move this and the next line outside of the if/else so that we don't need to repeat code at lines 1082 and 1083.